sdk/identity/azure_identity/src/process/mod.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore workdir
+
+use azure_core::{
+    credentials::AccessToken,
+    error::{Error, ErrorKind, Result},
+    process::Executor,
+};
+use std::{ffi::OsStr, marker::PhantomData, sync::Arc};
+
+use crate::env::Env;
+
+/// Runs a command in the appropriate platform shell and processes the output
+/// using the specified `OutputProcessor`.
+///
+/// - Windows: Runs `cmd /C {command}` in %SYSTEMROOT%
+/// - Everywhere else: Runs `/bin/sh -c {command}` in /bin
+pub(crate) async fn shell_exec<T: OutputProcessor>(
+    executor: Arc<dyn Executor>,
+    env: &Env,
+    command: &str,
+    _: PhantomData<T>,
+) -> Result<AccessToken> {
+    let (workdir, program, c_switch) = if cfg!(target_os = "windows") {
+        let system_root = env.var("SYSTEMROOT").map_err(|_| {
+            Error::message(
+                ErrorKind::Credential,
+                "SYSTEMROOT environment variable not set",
+            )
+        })?;
+        (system_root, "cmd", "/C")
+    } else {
+        ("/bin".to_string(), "/bin/sh", "-c")
+    };
```

Comment:
`cfg!()` still causes both branches of code to be compiled in. If there's no great reason to have both, just do this instead:

```rust
const fn params() -> (&'static str, &'static str, &'static str) {
    #[cfg(windows)]
    {
        ("1", "2", "3")
    }
    #[cfg(not(windows))]
    {
        ("a", "b", "c")
    }
}
let (workdir, program, c_switch) = params();

println!("{workdir}, {program}, {c_switch}");
```

There are some other ways of doing this, like just declaring a function with the same name outside the scope.